src/mr/worker.go

```diff
@@ -117,9 +121,11 @@ func Worker(mapf func(string, string) []KeyValue,
 			outFile.Close()
 		default:
 			isDone := CallIsMapReduceDone()
-			hasMoreWork = !isDone
+			hasMoreWork = !isDone  // could just use a break stmt here
 		}
 
+		// What happens if there is no task but the map/reduce job hasn't finished
+		// e.g. map phase is waiting for a single straggler so the orchestrator doesn't need to schedule a task on this node
```

Comment:
the loop handles that I think? If the coordinator returns false to `IsMapReduceDone` the loop will call the coordinator again asking for work